ocpmodels/trainers/forces_trainer.py

```diff
@@ -429,19 +438,33 @@ def train(self):
                             self.best_val_metric = val_metrics[primary_metric][
                                 "metric"
                             ]
-                            current_epoch = epoch + (i + 1) / len(
-                                self.train_loader
-                            )
-                            current_step = epoch * len(self.train_loader) + (
-                                i + 1
-                            )
                             self.save(current_epoch, current_step, val_metrics)
                             if self.test_loader is not None:
                                 self.predict(
                                     self.test_loader,
                                     results_file="predictions",
                                     disable_tqdm=False,
                                 )
+                        if self.is_hpo:
```

Comment:
The way this is placed means HPO is only accessible if `eval_every` is defined, and not the default per-epoch validation: https://github.com/Open-Catalyst-Project/ocp/pull/232/files#diff-b05769e2764b8b45457d8c3f0b7b203db2c100a8ab569849ae5174fc1a71f53dR474

I'm fine with this, someone can define the appropriate `eval_every` needed to run validation per epoch. But if this wasn't intended, you can update accordingly.

Reply:
Thanks for bringing this up @mshuaibii. Yeah, this was intended because for many models/datasets 1 epoch takes a long time to run and to fully leverage the benefits of HPO methods in Ray Tune we want to communicate val results on a much shorter time scale. But it needs to be documented I will add some directions to the HPO readme.